src/modules/ekf2/EKF/terrain_control.cpp

```diff
@@ -85,19 +77,17 @@ void Ekf::controlTerrainFakeFusion()
 
 bool Ekf::isTerrainEstimateValid() const
 {
-	// Assume being valid when the uncertainty is small compared to the height above ground
-	float hagl_var = INFINITY;
-	sym::ComputeHaglInnovVar(P, 0.f, &hagl_var);
-	bool valid = hagl_var < fmaxf(sq(0.1f * getHagl()), 0.2f);
-
-#if defined(CONFIG_EKF2_RANGE_FINDER)
-
-	// Assume that the terrain estimate is always valid when direct observations are fused
-	if (_control_status.flags.rng_terrain && isRecent(_aid_src_rng_hgt.time_last_fuse, (uint64_t)5e6)) {
-		valid = true;
+	if ((_control_status.flags.rng_terrain || _control_status.flags.opt_flow_terrain)
```

Comment:
When aiding stops the terrain state remains valid for some time, especially when the vehicle doesn't move horizontally; that's why I added "Assume being valid when the uncertainty is small compared to the height above ground" to run even when no aiding was active.

Reply:
Makes sense, I'll refactor/restore the logic here.